frontend/pages/SoftwarePage/SoftwareVulnerabilities/SoftwareVulnerabilities.tsx

```diff
@@ -66,23 +89,167 @@ const SoftwareVulnerabilities = ({
     () => getVulnerabilities(queryParams),
     {
       keepPreviousData: true,
-      staleTime: 30000,
+      enabled: !isExactMatchQuery && isSoftwareEnabled,
+      onSuccess: (data) => {
+        setTableData(data);
+        if (data.count === 0) {
+          if (
+            queryParams.exploit ||
+            (queryParams.query && queryParams.query.length > 0)
+          ) {
+            setEmptyStateReason("no-matching-items");
+          } else {
+            setEmptyStateReason("no-vulns-detected");
+          }
+        }
+      },
     }
   );
 
-  if (isLoading) {
+  // Calling software/vulnerabilities/:CVE endpoint when user searches with quotation marks
+  const {
+    isLoading: isLoadingExactMatch,
+    isFetching: isFetchingExactMatch,
+    refetch: refetchExactMatch,
+  } = useQuery<
+    IVulnerabilityResponse | null,
+    AxiosError<IApiError>,
+    IVulnerabilityResponse,
+    IGetVulnerabilityQueryKey[]
+  >(
+    [
+      {
+        scope: "softwareVulnByCVE",
+        vulnerability: (query && stripQuotes(query)) || "",
+        teamId,
+      },
+    ],
+    ({ queryKey }) => {
+      return softwareVulnAPI.getVulnerability(queryKey[0]);
+    },
+    {
+      ...DEFAULT_USE_QUERY_OPTIONS,
+      retry: false,
+      onSuccess: (data) => {
+        // Handle 204 response which doesn't return data if it is a known CVE but doesn't exist in response
+        if (!data) {
```

Comment:
It would be a bit more future-proof to check the status code directly if it's available but this also works.

Reply:
100% I did a quick wrestle this morning to see if it would be straightforward. Left a note in the code to revisit